pkg/providers/tinkerbell/validator.go

```diff
@@ -283,3 +295,22 @@ func hardwareMap(hardwareList []*tinkhardware.Hardware) map[string]*tinkhardware
 
 	return hardwareMap
 }
+
+// workflowMap returns all the workflows on the tinkerbell stack in the form of map with mac address as a key
+func workflowMap(workflowList []*tinkworkflow.Workflow) (map[string]*tinkworkflow.Workflow, error) {
+	workflowMap := make(map[string]*tinkworkflow.Workflow)
+
+	for _, data := range workflowList {
+		var macAddress map[string]string
+
+		if err := json.Unmarshal([]byte(data.GetHardware()), &macAddress); err != nil {
+			return nil, fmt.Errorf("error unmarshling workflow data: %v", err)
+		}
+
+		if _, ok := macAddress["device_1"]; ok {
```

Comment:
Just to be sure, this would always be `device _1` right? Even if we had different templates for CP and worker?

Reply:
I am not getting the mac address with the key now. So we are good here. Thanks for the suggestion.